### PR TITLE
Support semantic markup for roles

### PR DESCRIPTION
**Pull request by felixfontein**
Implements proposal from https://github.com/ansible-community/antsibull-docs/pull/113.

### CODE REVIEW

File: src/antsibull_docs_parser/parser.py [hunk 1]
```diff
@@ -165,10 +169,19 @@ def _parse_option_like(
         text = text[len(_IGNORE_MARKER) :]
     else:
         plugin_identifier = context.current_plugin
+        entrypoint = context.role_entrypoint
+    if plugin_identifier is not None and plugin_identifier.type == "role":
+        idx = text.find(":")
+        if idx >= 0:
+            entrypoint = text[:idx]
+            text = text[idx + 1 :]
+        if entrypoint is None:
```
Comment (gotmax23) [inline src/antsibull_docs_parser/parser.py:178]: Should this be

```suggestion
        if not entrypoint:
```
? I assume you'd also want a value like `:abc` (`text[:idx]` would be an empty string) to cause an error.

File: src/antsibull_docs_parser/parser.py [hunk 2]
```diff
@@ -165,10 +169,19 @@ def _parse_option_like(
         text = text[len(_IGNORE_MARKER) :]
     else:
         plugin_identifier = context.current_plugin
+        entrypoint = context.role_entrypoint
+    if plugin_identifier is not None and plugin_identifier.type == "role":
+        idx = text.find(":")
+        if idx >= 0:
+            entrypoint = text[:idx]
+            text = text[idx + 1 :]
```
Comment (gotmax23) [inline src/antsibull_docs_parser/parser.py:177]: Can you use `.partition()` here? I think it's a bit cleaner, but that's up to you.

```suggestion
    if plugin_identifier is not None and plugin_identifier.type == "role":
        part1, sep, part2 = text.partition(":")
        if sep:
            entrypoint = part1
            text = part2
```

File: changelogs/fragments/9-semantic-markup-roles.yml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+minor_changes:
+  - "Add support for semantic markup in roles (https://github.com/ansible-community/antsibull-docs-parser/pull/9)."
+breaking_changes:
```
Comment (gotmax23) [inline changelogs/fragments/9-semantic-markup-roles.yml:3]: Isn't adding `entrypoint` to `OptionNamePart` and `ReturnValuePart` also a breaking change? It changes the order of positional arguments when instantiating the class and the option indexes in the tuple are different.

File: changelogs/fragments/9-semantic-markup-roles.yml [hunk 4]
```diff
@@ -0,0 +1,4 @@
...
+  - "Modify ``LinkProvider.plugin_option_like_link`` signature to include optional string argument ``entrypoint`` after ``plugin`` (https://github.com/ansible-community/antsibull-docs-parser/pull/9)."
```
Comment (gotmax23) [inline changelogs/fragments/9-semantic-markup-roles.yml:4]: `optional string` seems confusing. The type is `str | None`, but the argument still needs to be passed.

**Review by gotmax23 — COMMENTED** [review]
As I said, I'm not super familiar with this codebase, but here are some suggestions.

File: src/antsibull_docs_parser/dom.py [hunk 5]
```diff
@@ -94,6 +94,7 @@ class CodePart(NamedTuple):
 
 class OptionNamePart(NamedTuple):
     plugin: t.Optional[PluginIdentifier]
+    entrypoint: t.Optional[str]  # present iff plugin.type == 'role'
```
[the remark this answers is not in the document]
Comment (felixfontein) [reply src/antsibull_docs_parser/dom.py:97]: `iff` is correct (https://en.wikipedia.org/wiki/Iff).

File: src/antsibull_docs_parser/parser.py (hunk 1 above)
Comment (felixfontein) [reply src/antsibull_docs_parser/parser.py:178]: I don't know whether ansible-core allows empty entrypoints. (Considering how flexible roles are in general, I wouldn't be surprised if it does.)

File: changelogs/fragments/9-semantic-markup-roles.yml (hunk 3 above)
Comment (felixfontein) [reply changelogs/fragments/9-semantic-markup-roles.yml:3]: Improved in 6ffdc1f7ece584fa2d85d4c9251eadeba931d6cf.

**Comment by felixfontein** [comment]
@gotmax23 thanks for reviewing this!

**Pull request merged**